### ISSUE TITLE
Sky Map fork with telescope control

### ISSUE THREAD

**Issue by marcocipriani01** [sky-map-team/stardroid#398]
_(off topic, this is not an issue, nor a feature request. I just hope someone finds this useful! 😃)_

Hello,
I really like Sky Map, but I've always thought it misses some features:

- Telescope & astronomy equipment control
- Modern UI with Material Design
- HD object thumbnails in the map
- Quicker loading times and no deprecated code used

So... I've forked Sky Map and added full telescope control using the [INDI library](http://indilib.org/) (+new user interface)!
It later became a completely different project, with a new name and under GPL v3 license. If someone is interested in this app, here's the repository: [github.com/marcocipriani01/Telescope.Touch](https://github.com/marcocipriani01/Telescope.Touch)! ~Coming soon on the Google Play Store.~ **Edit: [link to Play Store](https://play.google.com/store/apps/details?id=io.github.marcocipriani01.telescopetouch)**

Kind regards,
Marco Cipriani

**Comment by jaydeetay** [comment]
Good for you Marco! I hope that app is a success and look forward to trying
it out once it's published.

John

**Comment by jaydeetay** [comment]
Oh - and thanks for the credit in your README, that was kind.

**Comment by TacoTheDank** [comment]
Yooo this is pretty sweet! I'll also try it out :)

**Comment by TacoTheDank** [comment]
Perhaps you could help upstream out and create a few PRs for some of your improvements 😄 no pressure though

**Comment by marcocipriani01** [comment]
I would love to, but you are gonna have a bad time reviewing all my changes... Here's a non-exhaustive list of the changes I've made to the "Sky Map core" of Telescope.Touch:

- Full port to AndroidX
- Merged all the useful [Sky Map pull requests](https://github.com/sky-map-team/stardroid/pulls)
- Updated every Gradle dependecy
- Gallery with RecyclerView
- Almost every activity layout has been improved or changed
- New assets and icons, HD planet images
- Zoom limit in renderer
- Removed a lot of useless and redundant code
- Switching from Java `Date` (deprecated) to `Calendar` everywhere
- Removed Google Guava in favor of Java 8 utilities
- Completely reformatted code (I don't like Google Coding Style, but I would switch back to your style in pull requests😄)
- Got rid of Google Analytics and Kotlin (I can revert them)

Let me know what you think and the changes you want in upstream

**Comment by jaydeetay** [comment]
I think it's fine for it to be a separate project. There's a precedent for
this with Indian Sky Map.  We may well have different ideas for what we
want to do and different levels of time commitment.

**Comment by marcocipriani01** [comment]
@jaydeetay I totally agree with you, they are different projects, hence the new name and license. Telescope.Touch is more focused on telescope control rather than map viewing, and most of the changes I've made are functional to what I need. However, if you want me to contribute in some part to Sky Map, I'm available 😀

**Comment by barbeau** [comment]
@marcocipriani01 Very cool project, can't wait to check it out! The one thing that makes me sad, though, is the switch to the GPL license for your fork. It makes it impossible to share code and improvements with others Apache v2.0 licensed projects 😢.

**Comment by marcocipriani01** [comment]
@barbeau The reason is that the app uses code from my project [IPARCOS](https://github.com/INDIForJava/IPARCOS) for the telescope control part, that I forked from an old GPL v3 repo years ago. This chain obliges me to keep the GPL.
However, if someday you want me to pull some Sky Map improvements into upstream, don't worry. The changes I made to the Sky Map core are "my work", which means that I can do basically whatever I want with it, even copy some code to an Apache v2.0 project like this!

**Comment by barbeau** [comment]
@marcocipriani01 ok, that's good! To make it clear to others which code is licensed under GPL and which is under Apache v2, could you add license headers to files accordingly?

**Comment by marcocipriani01** [comment]
@barbeau I've changed the [project license notice](https://github.com/marcocipriani01/Telescope.Touch/blob/master/LICENSE.md) and added license headers to every Java class. All the Sky Map code is now back under Apache 2.0, with the hope this will help other projects 😄

**Comment by barbeau** [comment]
@marcocipriani01 Awesome, thanks, that's greatly appreciated!!

**Comment by marcocipriani01** [comment]
Update: now available on the Google Play Store: [play.google.com/store/apps/details?id=io.github.marcocipriani01.telescopetouch](https://play.google.com/store/apps/details?id=io.github.marcocipriani01.telescopetouch)

**Comment by jaydeetay** [comment]
Looks great! Like the upgrades to the icons and images even though I can't
use the telescope functionality.

John

**Comment by marcocipriani01** [comment]
You can copy both icons and images if you want to. Some of them are Google Material Icons, I've drawn the rest in Inkscape. Gallery images and planet thumbnails are from Hubble just like in Sky Map.
Regards,
Marco Cipriani

**Comment by jaydeetay** [comment]
Cheers!  I'm going to announce the app on our Facebook & Twitter channels
(don't get too excited: it's an audience of hundreds/thousands not
millions!).  Do you want me to mention you by name?

**Comment by marcocipriani01** [comment]
That sounds nice! I'll follow the channels😄 Yes, you can mention me by name

**Comment by marcocipriani01** [comment]
@jaydeetay thanks for the Tweet! The Play Store link you put doesn't work, however. Can you fix it?
https://play.google.com/store/apps/details?id=io.github.marcocipriani01.telescopetouch
(you only wrote "https://play.google.com/store/apps/details")

**Comment by jaydeetay** [comment]
Whoops. Must have been truncated. I'll fix it now.

**Comment by jaydeetay** [comment]
This is explicitly not an issue or feature request per the OP — an informational post about an external fork. Closing.
